qutebrowser/mainwindow/prompt.py

```diff
@@ -633,6 +635,42 @@ def __init__(self, question, parent=None):
         self._to_complete = ''
         self._root_index = QModelIndex()
 
+        self._history = cmdhistory.History(parent=self)
+        try:
+            fprompt_history = objreg.get('fprompt-history')
+            self._history.history = fprompt_history.data
+            self._history.changed.connect(fprompt_history.changed)
+            self._shortcut_next = QShortcut(QKeySequence("Ctrl+P"), self)
+            self._shortcut_next.activated.connect(self.history_next)
+            self._shortcut_prev = QShortcut(QKeySequence("Ctrl+N"), self)
+            self._shortcut_prev.activated.connect(self.history_prev)
+        except KeyError:
+            log.prompt.info("fprompt-history has not been initialized, "
+                            "no history completions available. "
+                            "Most likely running from pytest.")
```

Comment:
This seems quite odd. Please adjust the tests in a way they can deal with this without having to special case this in the code.

Reply:
I don't really know how to do this because if I try and run it normally it errors, if I try and run the function to init the history it errors because of the savemanager.

Reply:
There's a `fake_save_manager` [pytest fixture](https://docs.pytest.org/en/6.2.x/fixture.html) you can probably use.